### ISSUE TITLE
Magento adding blank rows to attribute options while trying to update through REST API

### ISSUE THREAD

**Issue by kingshukdeb** [magento/magento2#11272]
### Preconditions
----------------------
Magento version - 2.1.7 community and enterprise both
PHP - 5.6.25 and 7.0.2 both
MySql -  5.7.14 for community, not sure about the mysql version for EE

### Steps to reproduce

1.  First get the token from admin using REST call using this http://domain.com/rest/default/V1/integration/admin/token
2. Then using the token send another REST call to add/update the attribute like this http://domain.com/rest/V1/products/attributes/color (i am using color attribute here)
3.  Here is the whole code

```
<?php
		$userData = array("username" => "your username", "password" => "your password"); // Magento backend access details
		$ch = curl_init("http://domain.com/rest/default/V1/integration/admin/token"); // Direct hit
		curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "POST");
		curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($userData));
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
		//Specifying Accept type or by default JSON
		curl_setopt($ch, CURLOPT_HTTPHEADER, array("Content-Type: application/json", "Accept: application/json", "Content-Lenght: " . strlen(json_encode($userData))));
		$token = json_decode(curl_exec($ch));
		
		$ch = curl_init("http://domain.com/rest/V1/products/attributes/color");
		$attr = '{
		  "attribute": {
			"is_wysiwyg_enabled": false,
			"is_html_allowed_on_front": false,
			"used_for_sort_by": false,
			"is_filterable": false,
			"is_filterable_in_search": false,
			"is_used_in_grid": false,
			"is_visible_in_grid": false,
			"is_filterable_in_grid": false,
			"position": 0,
			"is_searchable": null,
			"is_visible_in_advanced_search": null,
			"is_comparable": "1",
			"is_used_for_promo_rules": null,
			"is_visible_on_front": "1",
			"used_in_product_listing": null,
			"is_visible": true,
			"scope": null,
			"attribute_id": 93,
			"attribute_code": "color",
			"frontend_input": "select",
			"entity_type_id": null,
			"is_required": false,
			"is_user_defined": false,
			"default_frontend_label": "Type of cloth",
			"frontend_labels": null,
			"backend_type": null,
			"is_unique": null,
			"options": [
			  {
				"label": "testinglabel",
				"value": "testingval",
				"sort_order": 0,
				"is_default": true,
				"store_labels": [
				  {
					"store_id": 0,
					"label": "testinglabelinside"
				  }
				]
			  }
			]
		  }
		}';
		
		// To use HTTPS without certificate
		curl_setopt ($ch, CURLOPT_SSL_VERIFYHOST, 0);
		curl_setopt ($ch, CURLOPT_SSL_VERIFYPEER, 0);
		curl_setopt($ch, CURLOPT_CUSTOMREQUEST, "PUT");
		curl_setopt($ch, CURLOPT_POSTFIELDS, $attr);
		curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
		curl_setopt($ch, CURLOPT_HTTPHEADER, array("Content-Type: application/json", "Authorization: Bearer ".$token));
		 
		$result = curl_exec($ch);
		print_r(json_decode($result));
```

### Expected result

1. 
![1](https://user-images.githubusercontent.com/4508615/31280059-bbc6acec-aac8-11e7-8596-be73c5da735c.jpg)
Its adding only 1 option to that attribute. This is fine.  But the issue is with this options param. If i have this param then its working fine. But if i dont have this param or just send "options":[] then its creating blank entries.

if i remove the below part then it will create a blank entries.
```
"options": [
			  {
				"label": "testinglabel",
				"value": "testingval",
				"sort_order": 0,
				"is_default": true,
				"store_labels": [
				  {
					"store_id": 0,
					"label": "testinglabelinside"
				  }
				]
			  }
			]
```
Or if i send like this then it will create blank entries.

`"options": []`

### Actual result
1.  Check here blank entries added after removing the options part and if i send options param like options[].
![2](https://user-images.githubusercontent.com/4508615/31280904-876a809c-aacb-11e7-83f3-359ecfeadefe.jpg)

**Comment by magento-engcom-team** [comment]
@kingshukdeb, thank you for your report.
The issue is already fixed in develop branch, 2.2.0, 2.1.9

**Comment by kingshukdeb** [comment]
I will check with 2.2.0 and will update. Thanks.
